controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -391,6 +406,14 @@ func (r *DSCInitializationReconciler) watchDSCResource(_ client.Object) []reconc
 
 		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: "backup"}}}
 	}
+	return nil
+}
 
+func (r *DSCInitializationReconciler) watchNamespaceResource(a client.Object) []reconcile.Request {
+	if !strings.HasPrefix(a.GetName(), "openshift-") && !strings.HasPrefix(a.GetName(), "kube-") &&
+		a.GetName() != "default" && a.GetName() != "openshift" {
+		r.Log.Info("Found a user namespace, start reconcile")
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: a.GetName()}}}
```

Comment:
Yes, probably it would be better to add reference to the DSCI instead of namespace.

So there are two things here that will trigger the configmap update:
1. Creation of new namespace
2. Update in DSCI `.spec.trustedCABundle` field.

I believe having a separate controller will not cover update in DSCI.
Let me know if I am missing something here

Reply:
I agree it will not, but we have captured it in another thread https://github.com/opendatahub-io/opendatahub-operator/pull/848#discussion_r1481530886

I still believe that would make sense to separate.

Reply:
Updated to introduce a new controller